R/s3exists.R

```diff
@@ -37,3 +35,15 @@ s3cmd_exists_command <- function(s3key) {
     paste("s3", "ls", s3key)
   }
 }
+
+
+check_exists_results <- function(name, results) {
+  ## We know that the key exists if a result was returned, i.e., the
+  ## shown regex gives a match.
+  if (use_legacy_api()) {
+    matches <- grepl(paste0(name, "(/[0-9A-Za-z]+)*/?$"), results)
+  } else {
+    matches <- grepl(paste0(basename(name), "$"), results)
```

Comment:
Neat! Hopefully `basename(name)` is never regex syntax that interferes with the grep. If it is, I say too bad -- should have picked a more sane key!